kernel/kernel-api/src/test/scala/io/delta/kernel/internal/util/PartitionUtilsSuite.scala

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util
+
+import java.util
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.expressions._
+import io.delta.kernel.expressions.Literal._
+import io.delta.kernel.internal.util.PartitionUtils.{rewritePartitionPredicateOnScanFileSchema, splitMetadataAndDataPredicates}
+import io.delta.kernel.types._
+import org.scalatest.funsuite.AnyFunSuite
+
+class PartitionUtilsSuite extends AnyFunSuite {
+  // Table schema
+  // Data columns: data1: int, data2: string, date3: struct(data31: boolean, data32: long)
+  // Partition columns: part1: int, part2: date, part3: string
+  private val partitionColsToType: java.util.Map[String, DataType] =
+  new util.HashMap[String, DataType]() {
+    {
+      put("part1", IntegerType.INSTANCE)
+      put("part2", DateType.INSTANCE)
+      put("part3", StringType.INSTANCE)
+    }
+  }
+
+  private val partitionCols: java.util.Set[String] = partitionColsToType.keySet()
+
+  // Test cases for verifying partition of predicate into data and partition predicates
+  // Map entry format (predicate -> (partition predicate, data predicate)
+  val partitionTestCases = Map[Predicate, (String, String)](
+    // single predicate on a data column
+    predicate("=", col("data1"), ofInt(12)) ->
+      ("ALWAYS_TRUE()", "(column(`data1`) = 12)"),
```

Comment:
nit: probably too late, but it should ideally just be
```scala
`column(data1)` 
```
instead of 
```scala
column(`data1`)
```

right?

i.e. we should only use back ticks for columns with `.`s in them?
```
column(a.b.`col.with.dot`.e)
```

Isn't this what spark does?

Reply:
It is just a `toString` method. If I recall correctly Delta-Spark does the same (add `` for every column). I can make it just adding backtick for names containing `.`, not sure if it is worth it.

Reply:
Cool. I just want to do what Spark does.